### ISSUE TITLE
Http call in unit test

### ISSUE THREAD

**Issue by aateeque** [cake-build/cake#1424]
### What You Are Seeing?
Failing unit test which attempt to make a web call as I am behind corp proxy

### What is Expected?
Unit tests should not make web calls

### What version of Cake are you using?
On the main branch

### Are you running on a 32 or 64 bit system?
64-bit

### What environment are you running on?  Windows? Linux? Mac?
Windows

### How Did You Get This To Happen? (Steps to Reproduce)
- Run tests using `dotnet test`
- Failing test: 
` Cake.Common.Tests.Unit.XML.XmlPeekAliasesTests+TheXmlPeekMethod.Should_Get_Node_Value_From_File_With_Dtd [FAIL]` 
- 

### Stack trace
```c#
  Starting:    Cake.Common.Tests
    Cake.Common.Tests.Unit.XML.XmlPokeTests+Transform.Should_Throw_On_Net_Core_Change_Attribute_From_Xml_File_With_Dtd [SKIP]
      .NET Core test.
    Cake.Common.Tests.Unit.XML.XmlPeekAliasesTests+TheXmlPeekMethod.Should_Get_Node_Value_From_File_With_Dtd [FAIL]
      System.Xml.XmlException : An error has occurred while opening external DTD 'http://www.apple.com/DTDs/PropertyList-1.0.dtd': The remote server returned an error: (407) Proxyentication Required.
      ---- System.Net.WebException : The remote server returned an error: (407) Proxy Authentication Required.
      Stack Trace:
           at System.Xml.XmlTextReaderImpl.Throw(Exception e)
           at System.Xml.XmlTextReaderImpl.PushExternalEntityOrSubset(String publicId, String systemId, Uri baseUri, String entityName)
           at System.Xml.XmlTextReaderImpl.DtdParserProxy_PushExternalSubset(String systemId, String publicId)
           at System.Xml.DtdParser.ParseExternalSubset()
           at System.Xml.DtdParser.ParseInDocumentDtd(Boolean saveInternalSubset)
           at System.Xml.DtdParser.Parse(Boolean saveInternalSubset)
           at System.Xml.DtdParser.System.Xml.IDtdParser.ParseInternalDtd(IDtdParserAdapter adapter, Boolean saveInternalSubset)
           at System.Xml.XmlTextReaderImpl.ParseDtd()
           at System.Xml.XmlTextReaderImpl.ParseDoctypeDecl()
           at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
           at System.Xml.XmlLoader.LoadDocSequence(XmlDocument parentDoc)
           at System.Xml.XmlDocument.Load(XmlReader reader)
        src\Cake.Common\Xml\XmlPeekAliases.cs(128,0): at Cake.Common.Xml.XmlPeekAliases.XmlPeek(XmlReader source, String xpath, XmlPeekSettings settings)
        src\Cake.Common\Xml\XmlPeekAliases.cs(93,0): at Cake.Common.Xml.XmlPeekAliases.XmlPeek(ICakeContext context, FilePath filePath, String xpath, XmlPeekSettings settings)
        src\Cake.Common.Tests\Fixtures\XmlPeekAliasesFixture.cs(45,0): at Cake.Common.Tests.Fixtures.XmlPeekAliasesFixture.Peek(String xpath)
        src\Cake.Common.Tests\Unit\XML\XmlPeekAliasesTests.cs(103,0): at Cake.Common.Tests.Unit.XML.XmlPeekAliasesTests.TheXmlPeekMethod.Should_Get_Node_Value_From_File_With_Dtd()
        ----- Inner Stack Trace -----
           at System.Net.HttpWebRequest.GetResponse()
           at System.Xml.XmlDownloadManager.GetNonFileStream(Uri uri, ICredentials credentials, IWebProxy proxy, RequestCachePolicy cachePolicy)
           at System.Xml.XmlUrlResolver.GetEntity(Uri absoluteUri, String role, Type ofObjectToReturn)
           at System.Xml.XmlTextReaderImpl.OpenAndPush(Uri uri)
           at System.Xml.XmlTextReaderImpl.PushExternalEntityOrSubset(String publicId, String systemId, Uri baseUri, String entityName)
    Cake.Common.Tests.Unit.XML.XmlPokeTests+Transform.Should_Change_Attribute_From_Xml_File_With_Dtd [FAIL]
      System.Xml.XmlException : An error has occurred while opening external DTD 'http://www.apple.com/DTDs/PropertyList-1.0.dtd': The remote server returned an error: (407) Proxyentication Required.
      ---- System.Net.WebException : The remote server returned an error: (407) Proxy Authentication Required.
      Stack Trace:
           at System.Xml.XmlTextReaderImpl.Throw(Exception e)
           at System.Xml.XmlTextReaderImpl.PushExternalEntityOrSubset(String publicId, String systemId, Uri baseUri, String entityName)
           at System.Xml.XmlTextReaderImpl.DtdParserProxy_PushExternalSubset(String systemId, String publicId)
           at System.Xml.DtdParser.ParseExternalSubset()
           at System.Xml.DtdParser.ParseInDocumentDtd(Boolean saveInternalSubset)
           at System.Xml.DtdParser.Parse(Boolean saveInternalSubset)
           at System.Xml.DtdParser.System.Xml.IDtdParser.ParseInternalDtd(IDtdParserAdapter adapter, Boolean saveInternalSubset)
           at System.Xml.XmlTextReaderImpl.ParseDtd()
           at System.Xml.XmlTextReaderImpl.ParseDoctypeDecl()
           at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
           at System.Xml.XmlLoader.LoadDocSequence(XmlDocument parentDoc)
           at System.Xml.XmlDocument.Load(XmlReader reader)
        src\Cake.Common\Xml\XmlPokeAliases.cs(510,0): at Cake.Common.Xml.XmlPokeAliases.XmlPoke(XmlReader source, XmlWriter destination, String xpath, String value, XmlPokeSettingtings)
        src\Cake.Common\Xml\XmlPokeAliases.cs(249,0): at Cake.Common.Xml.XmlPokeAliases.XmlPoke(ICakeContext context, FilePath filePath, String xpath, String value, XmlPokeSettingtings)
        src\Cake.Common.Tests\Fixtures\XmlPokeFixture.cs(48,0): at Cake.Common.Tests.Fixtures.XmlPokeFixture.Poke(String xpath, String value)
        src\Cake.Common.Tests\Unit\XML\XmlPokeTests.cs(124,0): at Cake.Common.Tests.Unit.XML.XmlPokeTests.Transform.Should_Change_Attribute_From_Xml_File_With_Dtd()
        ----- Inner Stack Trace -----
           at System.Net.HttpWebRequest.GetResponse()
           at System.Xml.XmlDownloadManager.GetNonFileStream(Uri uri, ICredentials credentials, IWebProxy proxy, RequestCachePolicy cachePolicy)
           at System.Xml.XmlUrlResolver.GetEntity(Uri absoluteUri, String role, Type ofObjectToReturn)
           at System.Xml.XmlTextReaderImpl.OpenAndPush(Uri uri)
           at System.Xml.XmlTextReaderImpl.PushExternalEntityOrSubset(String publicId, String systemId, Uri baseUri, String entityName)
  Finished:    Cake.Common.Tests
```

**Comment by patriksvensson** [comment]
I didn't know that the `XmlDocument.Load` was touching the network. We should fix this.

**Comment by devlead** [comment]
Fixed by #1452

**Comment by aateeque** [comment]
Didn't know about this web request by `XmlDocument.Load` either...probably the only time corp firewalls been useful :)  thanks @devlead!
